src/types/privatekubectl.ts

```diff
@@ -0,0 +1,143 @@
+import {Kubectl} from './kubectl'
+import {ExecOptions, ExecOutput, getExecOutput} from '@actions/exec'
+import * as core from '@actions/core'
+import * as os from 'os'
+import * as fs from 'fs'
+
+export class PrivateKubectl extends Kubectl {
+   public isPrivate(): boolean {
+      return true
+   }
+
+   protected async execute(args: string[], silent: boolean = false) {
+      args.unshift('kubectl')
+      let kubectlCmd = args.join(' ')
+      let addFileFlag = false
+      let eo = <ExecOptions>{silent}
+
+      if (this.containsFilenames(kubectlCmd)) {
+         core.debug('kubectl command contains filenames: ' + kubectlCmd)
+         // For private clusters, files will not be in the tmp directory
+         kubectlCmd = kubectlCmd.replace(/[\/][t][m][p][\/]/g, '')
```

Comment:
This is a bit flimsy in terms of maintainability. Could we use the path.basename function to decouple the directory name?

Reply:
I'm not following you. Kubectlcmd comes in with all of the filenames as /tmp/something... which is fine for public clusters. for private clusters the command gets executed from a different dir, and we remove the tmp prefix. The manifests can not exist in any directory where there are other files, otherwise invoke tries to process every file in the directory.

Reply:
This is stripping the /tmp/ from a filepath. Using path.basename does the same thing and it's not hardcoded to /tmp/.

Reply:
ahhhh,  got it